src/runtime_src/core/pcie/windows/alveo/device_windows.cpp

```diff
@@ -1187,6 +1197,86 @@ struct mailbox
   }
 }; //end of struct mailbox
 
+
+struct aim_counter
+{
+  using result_type = query::aim_counter::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::aim_counter::debug_ip_data_type>(arg1);
+
+    return xrt_core::debug_ip::get_aim_counter_result(device, dbg_ip_data);
+  }
+};
+
+struct am_counter
+{
+  using result_type = query::am_counter::result_type;
```

Comment:
Already defined in the headers?

Reply:
As per @stsoe  comment, he is going to look into these suggestions applicable for all of the device query methods and related data structures.